### DISCUSSION TITLE
Route collision!

### DISCUSSION THREAD

**Discussion by melvin2016** [vercel/next.js#21612] [Help]
I have a page called `home.js` in the root of pages folder and another one called `[username].js`. When I link to `/home` using the `Link` component, the `[username].js` route is capturing route instead of going to `home.js`.

Can someone help me on this?

**Comment by mkrtchian** [comment]
As [stated in the routing documentation](https://nextjs.org/docs/routing/dynamic-routes#caveats), predefined routes take precedence over dynamic routes. I just tried with next 10.0.6 and the same page names as yours, and it works as expected.
